### PR TITLE
Ticket6079 add ioc aeroflex

### PR DESCRIPTION
**Pull request by Scott-James-Hurley**
### Description of work

Adds OPI for Aeroflex signal generators. Allows user to set carrier frequency, RF level and modulation type as well as read these values from the device. Design is as similar as possible to the Aeroflex LabView GUIs.

### Ticket

ISISComputingGroup/IBEX#6079

### Acceptance criteria

Values can be set and changes are reflected in the readouts.

### System tests

Tested manually.

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] If the change is to an OPI, does the `check_opi_format.py` script in [C:\Instrument\Dev\ibex_gui\base\uk.ac.stfc.isis.ibex.opis](https://github.com/ISISComputingGroup/ibex_gui/blob/master/base/uk.ac.stfc.isis.ibex.opis/check_opi_format.py) pass?
- [ ] Do the changes function as described and is it robust?
- [ ] Is there associated PR for the [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)?

### Final Steps
- [ ] Reviewer has also merged the [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.opis/resources/opi_info.xml [hunk 1]
```diff
@@ -1,6 +1,24 @@
 <?xml version='1.0' encoding='UTF-8' standalone='yes'?>
 <descriptions>
   <opis>
+	<entry>
+      <key>Aeroflex</key>
+      <value>
+        <type>FUNCTION_GENERATOR</type>
+        <path>aeroflex/aeroflex.opi</path>
```
Comment (NikolaRoev) [inline base/uk.ac.stfc.isis.ibex.opis/resources/opi_info.xml:8]: If the python script is removed the OPI file can probably be taken out from the folder.

**Review by NikolaRoev — CHANGES_REQUESTED** [review]
- This OPI looks a lot of different from existing ones. Is there a reason for that? When creating a new OPI an easy way to setup widgets is to just copy them from `template.opi`.
- The OPI is missing the device name header, as well as the `NAME` macro. Check the template and existing OPIs.
- Should the current device type be displayed somewhere on the OPI? A PV that returns the `DEV_TYPE` macro can just be added.
- Is the grouping container `Settings` needed?
- Should the `Reset settings` button be renamed to `Factory settings` or something similar, as that will reset everything on the device not just the parameters exposed in the OPI?
- Is the python script for the frequency units needed? Just adding an additional units label should make the OPI much simpler. The DG645 OPI has an example of this.
- Do the readout parameter labels need `Actual` in front?
- The error message box feels a little big for the size of messages we might get on a real device.

**Comment by Scott-James-Hurley** [comment]
Users have expressed a desire for the IBEX GUI to remain similar to the existing LabView one
image

**Comment by Scott-James-Hurley** [comment]
image
Updated GUI

File: base/uk.ac.stfc.isis.ibex.opis/resources/opi_info.xml (hunk 1 above)
Comment (Scott-James-Hurley) [reply base/uk.ac.stfc.isis.ibex.opis/resources/opi_info.xml:8]: - I don't see the harm in keeping the 'Settings' grouping. It adds a symmetry that I think looks better than not having it
- I've no strong feelings about keeping `Actual` but it is in the LabView GUI
- I removed the units from the Carrier Frequency readout as the dropdown tells the user what units the reading is in

**Comment by NikolaRoev** [comment]
That looks great. I would just advise you squish everything, there is a lot of empty space still.

Should the `Actual carrier frequency` have units next to the value, like the previous iteration of the OPI or do you think that the combo box is enough?

**Comment by Scott-James-Hurley** [comment]
image

**Review by NikolaRoev — APPROVED** [review]

**Pull request merged**